crates/core/src/operations/optimize.rs

```diff
@@ -608,7 +606,7 @@ impl MergePlan {
         #[allow(unused_variables)] // used behind a feature flag
         max_spill_size: usize,
         min_commit_interval: Option<Duration>,
-        app_metadata: Option<HashMap<String, serde_json::Value>>,
+        commit_properties: CommitProperties,
```

Comment:
Why not make it here mutual as well?

Reply:
Hi Ion, I understand what you mean. Can you expand you what mean by mutual?

Reply:
Meant why not make the commit_properties mutable, like you did here and other places: https://github.com/delta-io/delta-rs/pull/2154/files#diff-a51cc3d9c5abab69a21a4d348682a89d656a59501bb3d2be0f24c0dc7b5b4bb6R193